src/SdssCentroid.cc

```diff
@@ -434,14 +437,27 @@ void SdssCentroidAlgorithm::measure(afw::table::SourceRecord &measRecord,
     int binX = 1;
     int binY = 1;
     double xc = 0., yc = 0., dxc = 0., dyc = 0.;  // estimated centre and error therein
+    bool stopBinning = false;
     for (int binsize = 1; binsize <= _ctrl.binmax; binsize *= 2) {
         std::tuple<MaskedImageT, double, int> smoothResult =
             smoothAndBinImage(psf, x, y, mimage, binX, binY, _flagHandler);
         int errorFlag = std::get<2>(smoothResult);
-        if (errorFlag > 0) {
+        if (errorFlag == EDGE.number) {
+            psf = std::make_shared<afw::detection::GaussianPsf>(5, 5, 0.5);
+            smoothResult = smoothAndBinImage(psf, x, y, mimage, binX, binY, _flagHandler);
+            stopBinning = true;
+            errorFlag = std::get<2>(smoothResult);
+            if (errorFlag == 0) {
+                errorFlag = NEAR_EDGE.number;
+            }
+        }
+        if (errorFlag > 0 && errorFlag != NEAR_EDGE.number) {
             _flagHandler.setValue(measRecord, errorFlag, true);
-            _flagHandler.setValue(measRecord, SdssCentroidAlgorithm::FAILURE.number, true);
-            return;
+            // if NEAR_EDGE is not a fatal error we continue otherwise return
+            if (errorFlag != NEAR_EDGE.number) {
+                _flagHandler.setValue(measRecord, SdssCentroidAlgorithm::FAILURE.number, true);
+                return;
+            }
```

Comment:
I think it's more conservative to keep considering this a FAILURE until we have evidence from processing that the fallback is not too bad.